bricks-maven-plugin/src/main/java/com/github/ingogriebsch/bricks/maven/plugin/analyzer/AbstractMavenAnalyzer.java

```diff
@@ -0,0 +1,63 @@
+/*-
+ * #%L
+ * Bricks Maven Plugin
+ * %%
+ * Copyright (C) 2018 - 2019 Ingo Griebsch
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.github.ingogriebsch.bricks.maven.plugin.analyzer;
+
+import com.github.ingogriebsch.bricks.model.Component;
+
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.project.MavenProject;
+import org.reflections.Reflections;
+
+import lombok.NonNull;
+
+/**
+ * Optional convenience superclass that initializes fields from the context, so
+ * that subclasses can be less verbose.
+ * 
+ * 
+ * @author doc
+ *
+ */
+public abstract class AbstractMavenAnalyzer implements MavenAnalyzer {
+
+    protected Object session;
+
+    protected MavenProject project;
+
+    protected Log log;
+
+    protected AnalyzerContext ctx;
+
+    protected Reflections reflections;
+
+    @Override
+    public final AnalysisResult augment(@NonNull AnalyzerContext ctx, @NonNull Component c) throws Exception {
+
```

Comment:
Empty line?!

Reply:
why are empty lines bad? we use it to make the structure of the code obvious and easier to grasp. every single usage of empty lines is probably debateable (this on especially), but i'd like to know the motivation behind your hunt for empty lines, as it might be something i missed up to now.

Reply:
How you are using empty lines look like you are following a pattern (put an empty line behind every method definition and before the closing brace, which I don't understand/find nice. If so, would be something for the formatter and should be followed always.

In general I really like to group/structure the code to make it more readable (as you can see in this PR because I did not comment every empty line). ;)